### DISCUSSION TITLE
How To Get A pProjector In The doFDK()?

### DISCUSSION THREAD

**Discussion by dyxer** [astra-toolbox/astra-toolbox#474] [Q&A]
Hello everyone!
I'm trying to reconstruct a 3D battery  with 360 2D projection images in the linux system. i had installed astra, got its header files and shared library file libastra.so. I use the doFDK() algoriyhm,its define likes this : doFDK(CProjector3D *pProjector, CFloat32VolumeData3D *pVolData,
	          CFloat32ProjectionData3D *pProjData, bool bShortScan,
	          const float *pfFilter = 0, SJob::EMode eMode = SJob::MODE_SET) .
in the codes, 
vol_geom = createVolGeom(voxel_size,xres,yres,zres,0.0,0.0,0.0);
proj_geom = createConeProjGeom(&cthdr);
CProjector3D* pProjector=new CProjector3D();
volumeData = create_data_3d_vol(vol_geom,0.0f);
sinoData = load_data_3d_sino(file_sino,proj_geom);
CCompositeGeometryManager *cgeomgr;
cgeomgr->doFDK(pPeojector,volumeData,sinoData,false,NULL);
 The question is that I don't know what and how to use to initialize this pPeojector.Would you have any idea how to achieve this within ASTRA? Thanks a lot!

**Comment by askorikov** [comment]
Hi! C++ API for ASTRA is not officially supported, but you can have a look at #110 (and more specifically here: https://gitlab.com/seirios/astra-toolbox/-/tree/cpp/cpp).
